src/renderer/decoder/common.js

```diff
@@ -0,0 +1,92 @@
+import { getJoinNormal, getLineNormal, neg } from '../../utils/geometry';
+
+/**
+ * Create a triangulated lineString: zero-sized, vertex-shader expanded triangle list
+ * with `miter` joins. For angle < 60 joins are automatically adjusted to `bevel`.
+ */
+export function addLine (lineString, vertices, normals, isPolygon, skipCallback) {
+    let prevPoint, currentPoint, nextPoint;
+    let prevNormal, nextNormal;
+    let skipLine = false;
+
+    // We need at least two points
+    if (lineString.length >= 4) {
+        // Initialize the first two points
+        prevPoint = [lineString[0], lineString[1]];
+        currentPoint = [lineString[2], lineString[3]];
+        prevNormal = getLineNormal(prevPoint, currentPoint);
+
+        for (let i = 4; i <= lineString.length; i += 2) {
```

Comment:
The `length` micro-optimization has been removed from here too.

Reply:
https://github.com/CartoDB/carto-vl/pull/791#discussion_r207173054